statsmodels/stats/control_chart.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+"""Control Charts and Statistical Process Control
+
+Created on Sun Dec 31 23:47:52 2017
+
+Author: Josef Perktold
+
+
+Some tools for building control charts and process monitoring.
+The core should be tools for the statistics, estimation and control limits.
+Control charts and monitoring interface are just for a basic example use case.
+
+"""
+
+from __future__ import division
+import numpy as np
+from scipy import stats
+import matplotlib.pyplot as plt
+
+def update_std(std, nobs, y_new):
+    raise NotImplementedError
+
+class Holder(object):
+    def __init__(self, **kwds):
+        self.__dict__.update(kwds)
+
+def control_limits_mean(y0, alpha=0.05, crit=None, distr='t'):
+
+        endog = y0
+        nobs = endog.shape[0]
+        #index = np.arange(nobs)  # not used nor returned
+
+        if crit is not None:
+            crit = crit
+        else:
+            if distr == 't':
+                crit = stats.t.isf(alpha/2, nobs-1)
+            elif distr == 'n':
+                crit = stats.norm.isf(alpha/2)
+            else:
+                raise ValueError('only t and normal for now' )
+
+        center = endog.mean(0)
+        std = endog.std(0, ddof=1)
+        diff = crit * std
+        upp = center + diff
+        low = center - diff
+        res = Holder(center=center,
+                     std=std,
+                     low=low,
+                     upp=upp,
+                     diff=diff)
+        return res
+
+
+class ControlChartMean(object):
+
+    def __init__(self, y0, alpha=0.05, crit=None, distr='t'):
+
+        self.endog = y0
+        self.nobs = self.endog.shape[0]
+        self.index = np.arange(self.nobs)
+
+        if crit is not None:
+            self.crit = crit
+        else:
+            if distr == 't':
+                self.crit = stats.t.isf(alpha/2, self.nobs-1)
+            elif distr == 'n':
+                self.crit = stats.norm.isf(alpha/2)
+            else:
+                raise ValueError('only t and normal for now' )
+
+        self.center = self.endog.mean(0)
+        self.std = self.endog.std(0, ddof=1)
```

Comment:
same as above
center and std computation need to handle weights and use options, e.g. for robust estimators